src/plonk/keygen.rs

```diff
@@ -198,33 +199,76 @@ impl<F: Field> Assignment<F> for Assembly<F> {
     }
 }
 
+fn k_from_circuit<F: Ord + Field + FromUniformBytes<64>, C: Circuit<F>>(circuit: &C) -> u32 {
```

Comment:
This is repeating the code of keygen.
I think it's fine because it is very little code, but how about adding a comment to mention it?